docs/src/modules/components/HighlightedCode.js

```diff
@@ -1,30 +1,72 @@
+/* eslint-disable material-ui/no-hardcoded-labels */
 import * as React from 'react';
 import PropTypes from 'prop-types';
+import copy from 'clipboard-copy';
 import prism from '@mui/markdown/prism';
 import MarkdownElement from 'docs/src/modules/components/MarkdownElement';
+import { useCodeCopy } from 'docs/src/modules/utils/CodeCopy';
 
 const HighlightedCode = React.forwardRef(function HighlightedCode(props, ref) {
-  const { code, language, component: Component = MarkdownElement, ...other } = props;
+  const {
+    analytics,
+    code,
+    language,
+    component: Component = MarkdownElement,
+    enableCodeCopy,
+    ...other
+  } = props;
   const renderedCode = React.useMemo(() => {
     return prism(code.trim(), language);
   }, [code, language]);
+  const [copied, setCopied] = React.useState(false);
+  const handlers = useCodeCopy();
+  React.useEffect(() => {
+    if (copied) {
+      const timeout = setTimeout(() => {
+        setCopied(false);
+      }, 2000);
+      return () => {
+        clearTimeout(timeout);
+      };
+    }
+    return () => {};
+  }, [copied]);
 
   return (
-    <Component ref={ref} {...other}>
-      <pre>
-        <code
-          className={`language-${language}`}
-          // eslint-disable-next-line react/no-danger
-          dangerouslySetInnerHTML={{ __html: renderedCode }}
-        />
-      </pre>
+    <Component ref={ref} enableCodeCopy={enableCodeCopy} {...other}>
+      <div className="MuiCode-root" {...handlers}>
+        <pre>
+          <code
+            className={`language-${language}`}
+            // eslint-disable-next-line react/no-danger
+            dangerouslySetInnerHTML={{ __html: renderedCode }}
+          />
+        </pre>
+        {enableCodeCopy && (
+          <button
+            {...analytics}
+            aria-label="Copy the code"
+            type="button"
+            className="MuiCode-copy"
+            onClick={async (event) => {
+              event.stopPropagation();
```

Comment:
Why stop the propagation of the click?

Reply:
The button here could be added by other listeners but maybe I overcomplicated it. Will remove the `stopPropagation`.